app/services/events/record.rb

```diff
@@ -220,6 +220,16 @@ def self.record_induction_period_reopened_event!(author:, induction_period:, mod
       new(event_type:, induction_period:, modifications:, author:, appropriate_body:, teacher:, heading:, happened_at:, body:).record_event!
     end
 
+    # ECT and mentor events
+
+    def self.record_teacher_registered_as_mentor_event!(author:, mentor_at_school_period:, teacher:, school:, happened_at: Time.zone.now)
```

Comment:
We can get the :school from the : mentor_at_school_period.
Worth considering whether to remove the :school param even though that might trigger an extra query to the db?

Reply:
I'm on the fence with this, it makes things less verbose but means we need to set relationships up in the specs which has become a problem in the past.